src/libsyntax/ast.rs

```diff
@@ -1691,45 +1691,20 @@ pub struct Variant_ {
 
 pub type Variant = Spanned<Variant_>;
 
-#[derive(Clone, PartialEq, Eq, RustcEncodable, RustcDecodable, Hash, Debug, Copy)]
-pub struct PathListItem_ {
-    pub name: Ident,
-    /// renamed in list, e.g. `use foo::{bar as baz};`
-    pub rename: Option<Ident>,
-    pub id: NodeId,
-}
-
-pub type PathListItem = Spanned<PathListItem_>;
-
-pub type ViewPath = Spanned<ViewPath_>;
-
 #[derive(Clone, PartialEq, Eq, RustcEncodable, RustcDecodable, Hash, Debug)]
-pub enum ViewPath_ {
-
-    /// `foo::bar::baz as quux`
-    ///
-    /// or just
-    ///
-    /// `foo::bar::baz` (with `as baz` implicitly on the right)
-    ViewPathSimple(Ident, Path),
-
-    /// `foo::bar::*`
-    ViewPathGlob(Path),
-
-    /// `foo::bar::{a,b,c}`
-    ViewPathList(Path, Vec<PathListItem>)
+pub enum ViewPathKind_ {
+    ViewPathSimple(Ident),
+    ViewPathGlob,
+    ViewPathNested(Vec<(ViewPath, NodeId)>),
 }
 
-impl ViewPath_ {
-    pub fn path(&self) -> &Path {
-        match *self {
-            ViewPathSimple(_, ref path) |
-            ViewPathGlob (ref path) |
-            ViewPathList(ref path, _) => path
-        }
-    }
+#[derive(Clone, PartialEq, Eq, RustcEncodable, RustcDecodable, Hash, Debug)]
+pub struct ViewPath_ {
+    pub kind: ViewPathKind_,
+    pub path: Path,
 }
 
+pub type ViewPath = Spanned<ViewPath_>;
```

Comment:
Could you avoid `Spanned` here and add the `span: Span` field into `ViewPath_` explicitly, and then rename `ViewPath_` -> `ViewPath` and `ViewPathKind_` -> `ViewPathKind`.

Reply:
It's also a good opportunity to finally rename `ViewPath` (what does it even mean?) into something more reasonable, e.g. `ImportTree` or `UseTree`.